api/src/test/java/org/openmrs/api/AdministrationServiceTest.java

```diff
@@ -556,6 +560,208 @@ public void getGlobalProperty_shouldGetPropertyInCaseInsensitiveWay() {
+		property.setPropertyValue("test_property_value");
+		property.setViewPrivilege(Context.getUserService().getPrivilege("Some Privilege For View Global Properties"));
+		adminService.saveGlobalProperty(property);
+		// assert new test global property is saved properly
+		List<GlobalProperty> properties = adminService.getAllGlobalProperties();
+		assertEquals(expectedSize + 1, properties.size());
+		
+		if (Context.isAuthenticated()) {
+			Context.logout();
+		}
+		
+		// authenticate new user to test view permissions
+		Context.authenticate("test_user", "test");
+		// have to add privilege in order to be able to call getAllGlobalProperties() method for new user
+		Context.addProxyPrivilege(PrivilegeConstants.GET_GLOBAL_PROPERTIES);
+		
+		properties = adminService.getAllGlobalProperties();
+		int actualSize = properties.size();
+		
+		Context.removeProxyPrivilege(PrivilegeConstants.GET_GLOBAL_PROPERTIES);
+		Context.logout();
+		
+		assertEquals(actualSize, expectedSize);
+		assertTrue(!properties.contains(property));
+	}
+	
+	/**
+	 * @see org.openmrs.api.AdministrationService#getGlobalProperty(java.lang.String)
+	 */
+	@Test
+	public void getGlobalProperty_shouldFailIfUserHasNoPrivileges() {
+		executeDataSet(ADMIN_INITIAL_DATA_XML);
+		GlobalProperty property = getGlobalPropertyWithViewPrivilege();
+		
+		User user = Context.getUserService().getUserByUsername("test_user");
+		assertNotNull(user);
+		// left this user without required privilege
+		Context.becomeUser(user.getSystemId());
+		
+		APIException exception = assertThrows(APIException.class, () -> adminService.getGlobalProperty(property.getProperty()));
+		assertEquals(exception.getMessage(), String.format("Privilege: %s, required to view globalProperty: %s",
+			property.getViewPrivilege(), property.getProperty()));
+	}
+	
+	/**
+	 * @see org.openmrs.api.AdministrationService#getGlobalProperty(java.lang.String)
+	 */
+	@Test
+	public void getGlobalProperty_shouldReturnGlobalPropertyIfUserIsAllowedToView() {
+		executeDataSet(ADMIN_INITIAL_DATA_XML);
+		GlobalProperty property = getGlobalPropertyWithViewPrivilege();
+		
+		User user = Context.getUserService().getUserByUsername("test_user");
+		assertNotNull(user);
+		// add required privilege to user
+		Role role = Context.getUserService().getRole("Provider");
+		role.addPrivilege(property.getViewPrivilege());
+		user.addRole(role);
+		Context.becomeUser(user.getSystemId());
+		
+		assertNotNull(adminService.getGlobalProperty(property.getProperty()));
+	}
+	
+	/**
+	 * @see org.openmrs.api.AdministrationService#updateGlobalProperty(java.lang.String, java.lang.String)
+	 */
+	@Test
+	public void updateGlobalProperty_shouldFailIfUserIsNotAllowedToEditGlobalProperty() {
+		executeDataSet(ADMIN_INITIAL_DATA_XML);
+		GlobalProperty property = getGlobalPropertyWithEditPrivilege();
+		assertEquals("anothervalue", property.getPropertyValue());
+		
+		User user = Context.getUserService().getUserByUsername("test_user");
+		assertNotNull(user);
```

Comment:
What do you think about wrapping these three repetitive lines in a helper method?